src/javascript/app/pages/trade/charts/highchart.ui.js

```diff
@@ -25,19 +27,21 @@ const HighchartUI = (() => {
                 return `<div style="${common_spot_style} background-color: orange; width:10px; height: 10px;"></div> ${localize('Exit spot')} `;
             case 'end_time':
                 return `<div style="${common_time_style} border-style: dashed;"></div> ${localize('End time')} `;
+            case 'reset_time':
+                return `<div style="${reset_time_style}  border-style: solid;"></div> ${localize('Reset time')} `;
             case 'delay':
                 return `<span class="chart-delay"> ${localize('Charting for this underlying is delayed')} </span>`;
             default:
                 return null;
         }
     };
 
-    const setLabels = (chart_delayed) => {
+    const setLabels = (chart_delayed, is_reset_contract) => {
         // display a guide for clients to know how we are marking entry and exit spots
         txt = (chart_delayed ? getLabels('delay') : '') +
             getLabels('start_time') +
             (history ? getLabels('entry_spot') + getLabels('exit_spot') : '') +
-            getLabels('end_time');
+            getLabels('end_time') + (is_reset_contract ? getLabels('reset_time') : '');
```

Comment:
shouldn't reset time be in the middle of the legend? right now it's pretty chronological, if reset time occurs in the middle of entry spot and exit spot then we should put it there

Reply:
sure, that's a better idea